docker/package/tezos_voting_wizard.py

```diff
@@ -118,17 +133,42 @@ def get_proposal_period_hash(hashes):
 )
 
 
-def get_node_rpc_addr_query(default=None):
+def get_node_rpc_addr_query(network):
+    node_validator = reachable_url_validator("chains/main/blocks/head/header")
+
+    def get_node_rpc_addr_query_validator(nodes):
+        def helper(answer):
+            if answer.isdigit():
+                answer = int(answer)
+                if answer not in range(1, len(nodes) + 1):
+                    raise ValueError("Please choose one of the provided values")
+                return nodes[answer - 1][0]
+            else:
+                return node_validator(answer)
```

Comment:
This is very similar to `enum_range_validator` which already exists. Perhaps you can use something like `required_field_validator if relevant_nodes is None/<empty> else enum_range_validator(relevant_nodes)`? I haven't thought about this in depth, but seems like it should be possible